src/BlockEntities/EnderChestEntity.cpp

```diff
@@ -46,10 +48,36 @@ void cEnderChestEntity::SendTo(cClientHandle & a_Client)
 
 
 
+class cFindSittingCat :
+	public cEntityCallback
+{
+	virtual bool Item(cEntity * a_Entity) override
+	{
+		return (
+			(a_Entity->GetEntityType() == cEntity::etMonster) &&
+			(static_cast<cMonster *>(a_Entity)->GetMobType() == eMonsterType::mtOcelot) &&
+			(static_cast<cOcelot *>(a_Entity)->IsSitting())
+		);
+	}
+};
```

Comment:
Rather than copying the code into two places, you should refactor it into a separate static method in the `cOcelot` class, something like `static bool cOcelot::IsCatSittingOnTopOfBlock(a_World, a_BlockX, a_BlockY, a_BlockZ)`